File: 0036-valid-sudoku/0036-valid-sudoku.java
import java.util.HashSet;
import java.util.Set;

public class Solution {
    public boolean isValidSudoku(char[][] board) {
        Set<String> seen = new HashSet<>();
        
        for (int i = 0; i < 9; ++i) {
            for (int j = 0; j < 9; ++j) {
                char number = board[i][j];
                if (number != '.') {
                    String row = number + " seen in row " + i;
                    String col = number + " seen in col " + j;
                    String box = number + " seen in box " + i / 3 + "-" + j / 3;
                    
                    if (seen.contains(row) || seen.contains(col) || seen.contains(box)) {
                        return false;
                    }
                    
                    seen.add(row);
                    seen.add(col);
                    seen.add(box);
                }
            }
        }
        
        return true;
    }
}

File: 0036-valid-sudoku/0036-valid-sudoku.cs
using System;
using System.Collections.Generic;

public class Solution {
    public bool IsValidSudoku(char[][] board) {
        HashSet<string> seen = new HashSet<string>();
        
        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                char number = board[i][j];
                if (number != '.') {
                    string row = number + " seen in row " + i;
                    string col = number + " seen in col " + j;
                    string box = number + " seen in box " + i / 3 + "-" + j / 3;
                    
                    if (seen.Contains(row) || seen.Contains(col) || seen.Contains(box)) {
                        return false;
                    }
                    
                    seen.Add(row);
                    seen.Add(col);
                    seen.Add(box);
                }
            }
        }
        
        return true;
    }
}
